pkg/field/fields.go

```diff
@@ -78,6 +85,10 @@ func BoolField(name string, optional ...fieldOption) SchemaField {
 		field = o(field)
 	}
 
+	if field.Required {
+		panic(fmt.Sprintf("requiring %s of type %s does not make sense", field.FieldName, field.FieldType))
+	}
```

Comment:
@ggreer thoughts?

Reply:
Seems fine.

Reply:
cool, let me mark the flags as required in Cobra and I'll let you know when this is ready to go